packages/main/cypress/specs/RatingIndicator.cy.tsx

```diff
@@ -100,4 +101,186 @@ describe("RatingIndicator", () => {
 			cy.focused().should("contain", "Before");
 		});
 	});
+
+	describe("Rating Indicator general interaction", () => {
+		it("Tests basic rating indicator rendering", () => {
+			cy.mount(<RatingIndicator id="rating-indicator1" accessibleName="Hello World"></RatingIndicator>);
+
+			cy.get("#rating-indicator1")
+				.shadow()
+				.find(".ui5-rating-indicator-item")
+				.should("have.length", 5);
+		});
+
+		it("Tests max property", () => {
+			cy.mount(<RatingIndicator id="rating-indicator2" value={6} max={10}></RatingIndicator>);
+
+			cy.get("#rating-indicator2")
+				.shadow()
+				.find(".ui5-rating-indicator-item")
+				.should("have.length", 10);
+		});
+
+		it("Tests clicking on star", () => {
+			cy.mount(<RatingIndicator id="rating-indicator3" value={6} max={10}></RatingIndicator>);
+
+			cy.get("#rating-indicator3").as("ri");
+
+			cy.get("@ri")
+				.shadow()
+				.find(".ui5-rating-indicator-item")
+				.eq(2)
+				.realClick();
+
+			cy.get("@ri")
+				.should("have.attr", "value", "3");
+		});
+
+		it("Tests change event", () => {
+			let changeEventCount = 0;
+			cy.mount(
+				<>
+					<RatingIndicator id="rating-indicator4" value={6} max={8}></RatingIndicator>
+					<input id="changes"></input>
+				</>
+			);
+
+			cy.get("#rating-indicator4").as("ri");
+
+			cy.get("@ri")
+				.then(ratingIndicator => {
+					ratingIndicator.get(0).addEventListener("ui5-change", () => {
+						changeEventCount++;
```

Comment:
this is not the proper way to assert events firing, look other tests for reference

Reply:
I tried to use stub or spy but neither of them have a counter how many times it has been called. It has built in methods calledOne, twice, truce but I want to validate against bigger number for times called.  A possible workaround is to split the test into smaller parts and check immediately after an action that the stub has been called.

Reply:
Done